agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2625,8 +2613,49 @@ bool backhaul_manager::handle_slave_ap_metrics_response(ieee1905_1::CmduMessageR
         std::copy_n(response.metric.estimated_service_info_field.begin(),
                     response.metric.estimated_service_info_field.size(),
                     ap_metrics_response_tlv->estimated_service_info_field());
+
+        for (auto stat : response.sta_traffic_stats) {
+
+            auto sta_traffic_response_tlv =
+                cmdu_tx.addClass<wfa_map::tlvAssociatedStaTrafficStats>();
+
+            if (!sta_traffic_response_tlv) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvAssociatedStaTrafficStats>";
+            }
+
+            sta_traffic_response_tlv->sta_mac()              = stat.sta_mac;
+            sta_traffic_response_tlv->byte_sent()            = stat.byte_sent;
+            sta_traffic_response_tlv->byte_recived()         = stat.byte_recived;
+            sta_traffic_response_tlv->packets_sent()         = stat.packets_sent;
+            sta_traffic_response_tlv->packets_recived()      = stat.packets_recived;
+            sta_traffic_response_tlv->tx_packets_error()     = stat.tx_packets_error;
+            sta_traffic_response_tlv->rx_packets_error()     = stat.rx_packets_error;
+            sta_traffic_response_tlv->retransmission_count() = stat.retransmission_count;
+        }
+
+        for (auto link_metric : response.sta_link_metrics) {
+
+            auto sta_link_metric_response_tlv =
+                cmdu_tx.addClass<wfa_map::tlvAssociatedStaLinkMetrics>();
+
+            if (!sta_link_metric_response_tlv) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvAssociatedStaLinkMetrics>";
+            }
+
+            sta_link_metric_response_tlv->sta_mac() = link_metric.sta_mac;
+            if (!sta_link_metric_response_tlv->alloc_bssid_info_list(1)) {
+                LOG(ERROR) << "Failed alloc_bssid_info_list";
+            }
+
+            auto &sta_link_metric_response =
+                std::get<1>(sta_link_metric_response_tlv->bssid_info_list(0));
+            sta_link_metric_response = link_metric.bssid_info;
+        }
     }
 
+    //Clear the m_ap_metric_response vector after preparing response for the controller
+    m_ap_metric_response.clear();
```

Comment:
m_ap_metric_query is storing the slaves and the BSSIDs for which AP metrics have been requested, but it is not storing the MID.
When slaves reply, items in m_ap_metric_query are removed by BSSID, without considering MID. 
**Only after m_ap_metric_query is empty** is that AP Metrics Response message is sent to controller.

This mechanism might fail if agent receives an AP Metrics Query message before it has processed previous one and the BSSIDs in both messages do not match.

Reply:
We kind of decided to keep it as is for M1 and refactor later -> #1328